---
title: "Gulf Stream Microbes"
author: "Ariane L. Peralta, Mario E. Muscarella, ..."
date: "`r format(Sys.time(), '%d %B, %Y')`"
header-includes:
   - \usepackage{array}
   - \usepackage{graphics}
output: 
  pdf_document:
    fig_caption: true
---

# Project Description:

# Initial Setup
```{r results='hide', message=FALSE}
rm(list=ls()) 
setwd("~/GitHub/GulfStreamMicrobiomes/analyses/")

# Simple Functions
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
sem <- function(x){sd(na.omit(x))/sqrt(length(na.omit(x)))}
CV <- function(x, ...){(sd(x, na.rm = TRUE)/mean(x, na.rm = TRUE))*100}

# Source Code
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")
# Save Default Plot Settings
opar <- par(no.readonly = TRUE)  # Saves plot defaults

# Required Packages
require("vegan"); require("stats");require("RColorBrewer")
require("png"); require("grid");require("reshape"); require("car")
```

# Import Data File
```{r}
design <- read.csv("../data/GS_design.csv")

otu.in <- read.otu("../data/GS.bac.final.shared")

tax.in <- read.tax("../data/GS.bac.final.taxonomy", format = "rdp", tax.levels = 6)

```

# OTU Data Wrangling
```{r}
# Groupings
cyanos <- tax.in$OTU[grep("Cyano", tax.in$Phylum)] ### NEED TO FIX - currently no cyanos
heteros <- tax.in$OTU[grep("Cyano", tax.in$Phylum, invert = T)]

# Remove OTUs with less than two occurences across all sites
otus <- otu.in[, which(colSums(otu.in) >= 2)]

aa <- (rowSums(otus))
aa # GS_13 [2181 reads], GS_14 [2749 reads] - removed

# Remove sites with low coverage
low <- which(rowSums(otus) < 5000)
rowSums(otus)[low]
odd.sites <- row.names(otus)[low]
otus <- otus[-c(low), ]

# Remove odd sites from design
design.2 <- design[-c(low), ]

# Remove Inshore
inshore <- design$MicroNumber[which(design$Location == "Inshore")]
otus <- otus[-c(which(row.names(otus) %in% inshore)), ]
design.2 <- design.2[-c(which(design.2$MicroNumber %in% inshore)), ]
design.2$Location <- droplevels(design.2$Location)

# Make Presence Absence Matrix
dataPA <- (otus > 0) * 1
dataPA.c <- dataPA[, which(colnames(dataPA) %in% cyanos)] #no sequences ??? double check
dataPA.h <- dataPA[, which(colnames(dataPA) %in% heteros)]

# Make Relative Abundance Matrices
otus.c <- otus[, which(colnames(otus) %in% cyanos)]
otus.h <- otus[, which(colnames(otus) %in% heteros)]

dataREL <- otus
for(i in 1:dim(otus)[1]){
  dataREL[i,] <- otus[i,]/sum(otus[i,])
}

dataREL.c <- otus.c
for(i in 1:dim(otus.c)[1]){
  dataREL.c[i,] <- otus.c[i,]/sum(otus.c[i,])
}

dataREL.h <- otus.h
for(i in 1:dim(otus.h)[1]){
  dataREL.h[i,] <- otus.h[i,]/sum(otus.h[i,])
}

# Log Transform Relative Abundances
dataREL.log <- decostand(dataREL, method="log")
dataRELc.log <- decostand(dataREL.c, method="log")
dataRELh.log <- decostand(dataREL.h, method="log")

# Backup: Rel before split
#dataREL.c2 <- dataREL[, which(colnames(dataREL) %in% cyanos)]
#dataREL.h2 <- dataREL[, which(colnames(dataREL) %in% heteros)]
```

# Alpha Diversity
```{r}
# Rarefy Abundances (min abundance is 69080. We are sampling to 69000)
min(rowSums(otus))
GSdata.r <- rrarefy(otus, 69000)

# Fisher's Alpha
fisher <- fisher.alpha(GSdata.r)

# Species Richness
#richness <- rowSums((PWESdata.r >= 1))
richness <- rowSums((otus >= 1))

# Shannon Diversity
shannon <- diversity(GSdata.r, "shannon")

# Simpson's Evenness
simp.even <- apply(GSdata.r, 1, simp_even)

#Pielouâ€™s evenness
J <- shannon/log(specnumber(GSdata.r[,-c(1:1)]))

#combined richness, diversity, evenness
diversity <- cbind(design.2,richness,shannon,simp.even,J)
```

#Plot shannon diversity 
```{r Plot - Shannon Diversity, echo=TRUE}
require(ggplot2)

# Graphing Shannon Diversity
p <- ggplot(diversity, aes(x=Location, y=shannon, color=as.factor(Location)))+ scale_color_manual(name="Location", values=c("#333333", "#3399FF","#66CC00" ), labels = c("Epibionts", "GS NO_Sargassum","GS Sargassum")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")+facet_wrap(~Cruise_date)+facet_grid(. ~Cruise_date)
shannon<- p1 + theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line=element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"), axis.text= element_text(size=12),axis.text.x = element_text(vjust=0.65, hjust=1.0, angle=90, size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + 
    labs(x = "Location", y = "Shannon Diversity Index (H')") + 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y = element_text(size=14, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1)) +
    scale_x_discrete(breaks=c("Epibionts", "NO_Sargassum","Sargassum"), labels=c("Epi", "NO_Sarg","Sarg"))
shannon  

ggsave("../figures/shannon.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)

# Graphing Species Richness - 
p <- ggplot(diversity, aes(x=Location, y=richness, color=as.factor(Location)))+ scale_color_manual(name="Location", values=c("#333333", "#3399FF","#66CC00" ), labels = c("Epibionts", "GS NO_Sargassum","GS Sargassum")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")+facet_wrap(~Cruise_date)+facet_grid(. ~Cruise_date)
richness <-p1 + theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line=element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"), axis.text= element_text(size=12),axis.text.x = element_text(vjust=0.65, hjust=1.0, angle=90, size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + 
    labs(x = "Location", y = "OTU Richness") + 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y = element_text(size=14, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1)) +
    scale_x_discrete(breaks=c("Epibionts", "NO_Sargassum","Sargassum"), labels=c("Epi", "NO_Sarg","Sarg"))
richness  
ggsave("../figures/richness.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)

# Graphing Pielou's Evenness
p <- ggplot(diversity, aes(x=Location, y=J, color=as.factor(Location)))+ scale_color_manual(name="Location", values=c("#333333", "#3399FF","#66CC00" ), labels = c("Epibionts", "GS NO_Sargassum","GS Sargassum")) + stat_summary(fun.data=mean_cl_boot,size=0.75) 
p1=p+geom_smooth(method="lm")+facet_wrap(~Cruise_date)+facet_grid(. ~Cruise_date)
evenness<- p1 + theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line=element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"), axis.text= element_text(size=12),axis.text.x = element_text(vjust=0.65, hjust=1.0, angle=90, size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm")) + 
    labs(x = "Location", y = "Pielou's Evenness (J')") + 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y = element_text(size=14, face="bold"), strip.background = element_rect(colour="black", fill="white", size=1)) +
    scale_x_discrete(breaks=c("Epibionts", "NO_Sargassum","Sargassum"), labels=c("Epi", "NO_Sarg","Sarg"))
evenness  

ggsave("../figures/evenness.png", plot=last_plot(), device=NULL, path=NULL, scale=1, width=7, height=5, dpi=300, limitsize=TRUE)
```

# Beta Diversity 
```{r}
# Data Checks
all.equal(as.character(design.2$MicroNumber), rownames(dataRELh.log))

# Trts
table(design.2$Location)

# Distance Matrix
#dataRELh.dist <- vegdist(dataRELh.log, method="bray")
#dataRELc.dist <- vegdist(dataRELc.log, method="bray")
dataRELh.dist <- vegdist(dataREL.h, method="bray")
dataRELc.dist <- vegdist(dataREL.c, method="bray")

# PERMANOVA
adonis.h = adonis(dataRELh.dist ~ design.2$Location*design.2$Cruise_date, method = "bray", perm=1000)
adonis.h

adonis.c = adonis(dataRELc.dist ~ design.2$Location*design.2$Cruise_date, method = "bray", perm=1000)
adonis.c

# Principal Coordinates Analysis - NO SOURCE
pcoa.h <- cmdscale(dataRELh.dist, k=3, eig=TRUE, add=FALSE)
pcoa.c <- cmdscale(dataRELc.dist, k=3, eig=TRUE, add=FALSE)

explainvar1h <- round(pcoa.h$eig[1] / sum(pcoa.h$eig), 3) * 100
explainvar2h <- round(pcoa.h$eig[2] / sum(pcoa.h$eig), 3) * 100

explainvar1c <- round(pcoa.c$eig[1] / sum(pcoa.c$eig), 3) * 100
explainvar2c <- round(pcoa.c$eig[2] / sum(pcoa.c$eig), 3) * 100

```

```{r}
design.2$Location <- gsub("_", " ", design.2$Location)

# Heterotrophs
points <- cbind(as.data.frame(pcoa.h$points), design.2$Location, design.2$Cruise_date)
L.centroids <- melt(points, id=c("design.2$Location", "design.2$Cruise_date"), measure.vars = c("V1", "V2"))
centroids <- cast(L.centroids, variable ~ design.2$Location + design.2$Cruise_date, mean)
centroids.se <- cast(L.centroids, variable ~ design.2$Location + design.2$Cruise_date, se)
centroids.sd <- cast(L.centroids, variable ~ design.2$Location + design.2$Cruise_date, sd)

cent.dataframe <- t(data.frame(rbind(centroids[1,-1], centroids[2,-1],
                             centroids.sd[1,-1],centroids.sd[2,-1])))
colnames(cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
cent.treats <- rownames(cent.dataframe)
pcoa.col <- as.factor(sapply(strsplit(cent.treats, "_"), `[`, 1))
pcoa.shape <- as.factor(sapply(strsplit(cent.treats, "_"), `[`, 2))
                                  
df <- as.data.frame(cent.dataframe)
p <- ggplot(df, aes(x=V1, y=V2, colour=pcoa.col, shape = pcoa.shape,
                group = interaction(pcoa.col, pcoa.shape))) + theme_bw() 
p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank(), plot.title = element_text(hjust = 0.5)) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
  geom_point(size=5) + 
  ggtitle("Heterotrophic Bacteria") +  
  xlab(paste("PCoA Axis 1 (", explainvar1h, "%)", sep = "")) + 
  ylab(paste("PCoA Axis 2 (", explainvar2h, "%)", sep = "")) + 
  scale_color_manual(name="Location", values=c("#333333", "#3399FF","#66CC00" ), 
                     labels = c("Epibionts", "NO_Sargassum","Sargassum")) + 
  scale_shape_manual(name="Cruise Date", 
                     labels = c("11/1/17", "5/10/17", "5/17/17", "8/3/17"),
                     values = c(15, 16, 17, 18)) +
ggsave("../figures/PCoA_Heterotrophs.pdf", plot=last_plot(), device=NULL, 
       path=NULL, scale=1, width=NA, height=NA, dpi=900, limitsize=TRUE)

# Cyanos
points <- cbind(as.data.frame(pcoa.c$points), design.2$Location, design.2$Cruise_date)
L.centroids <- melt(points, id=c("design.2$Location", "design.2$Cruise_date"), measure.vars = c("V1", "V2"))
centroids <- cast(L.centroids, variable ~ design.2$Location + design.2$Cruise_date, mean)
centroids.se <- cast(L.centroids, variable ~ design.2$Location + design.2$Cruise_date, se)
centroids.sd <- cast(L.centroids, variable ~ design.2$Location + design.2$Cruise_date, sd)


cent.dataframe <- t(data.frame(rbind(centroids[1,-1], centroids[2,-1],
                             centroids.sd[1,-1],centroids.sd[2,-1])))
colnames(cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
cent.treats <- rownames(cent.dataframe)
pcoa.col <- as.factor(sapply(strsplit(cent.treats, "_"), `[`, 1))
pcoa.shape <- as.factor(sapply(strsplit(cent.treats, "_"), `[`, 2))
                                  
df <- as.data.frame(cent.dataframe)
p <- ggplot(df, aes(x=V1, y=V2, colour=pcoa.col, shape = pcoa.shape,
                group = interaction(pcoa.col, pcoa.shape))) + theme_bw() 
p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank(), plot.title = element_text(hjust = 0.5)) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
  geom_point(size=5) + 
  ggtitle("Cyanobacteria") +  
  xlab(paste("PCoA Axis 1 (", explainvar1c, "%)", sep = "")) + 
  ylab(paste("PCoA Axis 2 (", explainvar2c, "%)", sep = "")) +  
  scale_color_manual(name="Location", values=c("#333333", "#3399FF","#66CC00" ), 
                     labels = c("Epibionts", "NO_Sargassum","Sargassum")) + 
  scale_shape_manual(name="Cruise Date", 
                     labels = c("11/1/17", "5/10/17", "5/17/17", "8/3/17"),
                     values = c(15, 16, 17, 18)) +

ggsave("../figures/PCoA_Cyanobacteria.pdf", plot=last_plot(), device=NULL, 
       path=NULL, scale=1, width=NA, height=NA, dpi=900, limitsize=TRUE)
```