---
title: "Gulf Stream Microbes"
author: "Ariane L. Peralta, Mario E. Muscarella, ..."
date: "`r format(Sys.time(), '%d %B, %Y')`"
header-includes:
   - \usepackage{array}
   - \usepackage{graphics}
output: 
  pdf_document:
    fig_caption: true
---

# Project Description:

# Initial Setup
```{r results='hide', message=FALSE}
rm(list=ls()) 
setwd("~/GitHub/GulfStreamMicrobiomes/analyses/")

# Simple Functions
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
sem <- function(x){sd(na.omit(x))/sqrt(length(na.omit(x)))}
CV <- function(x, ...){(sd(x, na.rm = TRUE)/mean(x, na.rm = TRUE))*100}

# Source Code
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")
# Save Default Plot Settings
opar <- par(no.readonly = TRUE)  # Saves plot defaults

# Required Packages
require("vegan"); require("stats");require("RColorBrewer")
require("png"); require("grid");require("reshape"); require("car")
```

# Import Data File
```{r}
design <- read.csv("../data/GS_design.csv")

otu.in <- read.otu("../data/phylotype/GS.trim.contigs.trim.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.tx.shared", cutoff = "1")

tax.in <- read.tax("../data/phylotype/GS.trim.contigs.trim.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.tx.1.cons.taxonomy", format = "rdp", tax.levels = 6)


tax_raw <- read.delim("../data/phylotype/GS.trim.contigs.trim.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.tx.1.cons.taxonomy", header = F, skip = 1) 

```

# OTU Data Wrangling
```{r}
# Groupings
cyanos <- tax.in$OTU[grep("Cyano", tax.in$Phylum)]
heteros <- tax.in$OTU[grep("Cyano", tax.in$Phylum, invert = T)]

# Remove OTUs with less than two occurences across all sites
otus <- otu.in[, which(colSums(otu.in) >= 2)]

# Remove sites with low coverage
low <- which(rowSums(otus) < 50000)
rowSums(otus)[low]
odd.sites <- row.names(otus)[low]
otus <- otus[-c(low), ]

# Remove odd sites from design
design.2 <- design[-c(low), ]

# Make Presence Absence Matrix
dataPA <- (otus > 0) * 1
dataPA.c <- dataPA[, which(colnames(dataPA) %in% cyanos)]
dataPA.h <- dataPA[, which(colnames(dataPA) %in% heteros)]

# Make Relative Abundence Matrices
otus.c <- otus[, which(colnames(otus) %in% cyanos)]
otus.h <- otus[, which(colnames(otus) %in% heteros)]

dataREL <- otus
for(i in 1:dim(otus)[1]){
  dataREL[i,] <- otus[i,]/sum(otus[i,])
}

dataREL.c <- otus.c
for(i in 1:dim(otus.c)[1]){
  dataREL.c[i,] <- otus.c[i,]/sum(otus.c[i,])
}

dataREL.h <- otus.h
for(i in 1:dim(otus.h)[1]){
  dataREL.h[i,] <- otus.h[i,]/sum(otus.h[i,])
}

# Log Transform Relative Abundances
dataREL.log <- decostand(dataREL, method="log")
dataRELc.log <- decostand(dataREL.c, method="log")
dataRELh.log <- decostand(dataREL.h, method="log")

# Backup: Rel before split
dataREL.c2 <- dataREL[, which(colnames(dataREL) %in% cyanos)]
dataREL.h2 <- dataREL[, which(colnames(dataREL) %in% heteros)]
```

# Alpha Diversity
```{r}


```

# Beta Diversity 
```{r}
# Data Checks
all.equal(as.character(design.2$MicroNumber), rownames(dataRELh.log))


# Distance Matrix
dataRELh.dist <- vegdist(dataRELh.log, method="bray")
dataRELc.dist <- vegdist(dataRELc.log, method="bray")


# PERMANOVA
adonis = adonis(dataRELh.dist ~ Date2*Dispersal*Salinity, method = "bray", data = csi.full.ns, perm=1000)
adonis

# Principal Coordinates Analysis - NO SOURCE
CSI_pcoa1 <- cmdscale(sampleREL.dist1, k=3, eig=TRUE, add=FALSE)
  # Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
  # eig=TRUE returns eigenvalues; k = # of dimensions to calculate

explainvar1a <- round(CSI_pcoa1$eig[1] / sum(CSI_pcoa1$eig), 3) * 100
explainvar2a <- round(CSI_pcoa1$eig[2] / sum(CSI_pcoa1$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)

explainvar1a #17.3
explainvar2a #7.3





```