---
title: "Gulf Stream Microbes"
author: "Ariane L. Peralta, Mario E. Muscarella, ..."
date: "`r format(Sys.time(), '%d %B, %Y')`"
header-includes:
   - \usepackage{array}
   - \usepackage{graphics}
output: 
  pdf_document:
    fig_caption: true
---

# Project Description:

# Initial Setup
```{r results='hide', message=FALSE}
rm(list=ls()) 
setwd("~/GitHub/GulfStreamMicrobiomes/analyses/")

# Simple Functions
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
sem <- function(x){sd(na.omit(x))/sqrt(length(na.omit(x)))}
CV <- function(x, ...){(sd(x, na.rm = TRUE)/mean(x, na.rm = TRUE))*100}

# Source Code
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")
# Save Default Plot Settings
opar <- par(no.readonly = TRUE)  # Saves plot defaults

# Required Packages
require("vegan"); require("stats");require("RColorBrewer")
require("png"); require("grid");require("reshape"); require("car")
```

# Import Data File
```{r}
design <- read.csv("../data/GS_design.csv")

otu.in <- read.otu("../data/phylotype/GS.trim.contigs.trim.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.tx.shared", cutoff = "1")

tax.in <- read.tax("../data/phylotype/GS.trim.contigs.trim.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.tx.1.cons.taxonomy", format = "rdp", tax.levels = 6)


tax_raw <- read.delim("../data/phylotype/GS.trim.contigs.trim.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.tx.1.cons.taxonomy", header = F, skip = 1) 

```

# OTU Data Wrangling
```{r}
# Groupings
cyanos <- tax.in$OTU[grep("Cyano", tax.in$Phylum)]
heteros <- tax.in$OTU[grep("Cyano", tax.in$Phylum, invert = T)]

# Remove OTUs with less than two occurences across all sites
otus <- otu.in[, which(colSums(otu.in) >= 2)]

# Remove sites with low coverage
low <- which(rowSums(otus) < 50000)
rowSums(otus)[low]
odd.sites <- row.names(otus)[low]
otus <- otus[-c(low), ]

# Remove odd sites from design
design.2 <- design[-c(low), ]

# Remove Inshore
inshore <- design$MicroNumber[which(design$Location == "Inshore")]
otus <- otus[-c(which(row.names(otus) %in% inshore)), ]
design.2 <- design.2[-c(which(design.2$MicroNumber %in% inshore)), ]
design.2$Location <- droplevels(design.2$Location)

# Make Presence Absence Matrix
dataPA <- (otus > 0) * 1
dataPA.c <- dataPA[, which(colnames(dataPA) %in% cyanos)]
dataPA.h <- dataPA[, which(colnames(dataPA) %in% heteros)]

# Make Relative Abundence Matrices
otus.c <- otus[, which(colnames(otus) %in% cyanos)]
otus.h <- otus[, which(colnames(otus) %in% heteros)]

dataREL <- otus
for(i in 1:dim(otus)[1]){
  dataREL[i,] <- otus[i,]/sum(otus[i,])
}

dataREL.c <- otus.c
for(i in 1:dim(otus.c)[1]){
  dataREL.c[i,] <- otus.c[i,]/sum(otus.c[i,])
}

dataREL.h <- otus.h
for(i in 1:dim(otus.h)[1]){
  dataREL.h[i,] <- otus.h[i,]/sum(otus.h[i,])
}

# Log Transform Relative Abundances
dataREL.log <- decostand(dataREL, method="log")
dataRELc.log <- decostand(dataREL.c, method="log")
dataRELh.log <- decostand(dataREL.h, method="log")

# Backup: Rel before split
dataREL.c2 <- dataREL[, which(colnames(dataREL) %in% cyanos)]
dataREL.h2 <- dataREL[, which(colnames(dataREL) %in% heteros)]
```

# Alpha Diversity
```{r}


```

# Beta Diversity 
```{r}
# Data Checks
all.equal(as.character(design.2$MicroNumber), rownames(dataRELh.log))

# Trts
table(design.2$Location)

# Distance Matrix
dataRELh.dist <- vegdist(dataRELh.log, method="bray")
dataRELc.dist <- vegdist(dataRELc.log, method="bray")


# PERMANOVA
adonis.h = adonis(dataRELh.dist ~ design.2$Location, method = "bray", perm=1000)
adonis.c = adonis(dataRELc.dist ~ design.2$Location, method = "bray", perm=1000)


# Principal Coordinates Analysis - NO SOURCE
pcoa.h <- cmdscale(dataRELh.dist, k=3, eig=TRUE, add=FALSE)
pcoa.c <- cmdscale(dataRELc.dist, k=3, eig=TRUE, add=FALSE)

explainvar1h <- round(pcoa.h$eig[1] / sum(pcoa.h$eig), 3) * 100
explainvar2h <- round(pcoa.h$eig[2] / sum(pcoa.h$eig), 3) * 100

explainvar1c <- round(pcoa.c$eig[1] / sum(pcoa.c$eig), 3) * 100
explainvar2c <- round(pcoa.c$eig[2] / sum(pcoa.c$eig), 3) * 100

```

```{r}
scale_color_manual(name="Location", values=c("#333333", "#3399FF","#66CC00" ), labels = c("Epibionts", "NO_Sargassum","Sargassum"))


# Heterotrophs
points <- cbind(as.data.frame(pcoa.h$points), design.2$Location)
L.centroids <- melt(points, id="design.2$Location", measure.vars = c("V1", "V2"))
centroids <- cast(L.centroids, variable ~ design.2$Location, mean)
centroids.se <- cast(L.centroids, variable ~ design.2$Location, se)
centroids.sd <- cast(L.centroids, variable ~ design.2$Location, sd)

cent.dataframe <- t(data.frame(rbind(centroids[1,-1], centroids[2,-1],
                             centroids.sd[1,-1],centroids.sd[2,-1])))
colnames(cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
cent.treats <- rownames(cent.dataframe)
                                  
df <- as.data.frame(cent.dataframe)
p <- ggplot(df, aes(x=V1, y=V2, colour=cent.treats)) + theme_bw() 
p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
  geom_point(size=5) +
  xlab(paste("PCoA Axis 1 (", explainvar1h, "%)", sep = "")) + 
  ylab(paste("PCoA Axis 2 (", explainvar2h, "%)", sep = "")) + 
  scale_color_manual(name="Location", values=c("#333333", "#3399FF","#66CC00" ), labels = c("Epibionts", "NO_Sargassum","Sargassum"))


# Cyanos
points <- cbind(as.data.frame(pcoa.c$points), design.2$Location)
L.centroids <- melt(points, id="design.2$Location", measure.vars = c("V1", "V2"))
centroids <- cast(L.centroids, variable ~ design.2$Location, mean)
centroids.se <- cast(L.centroids, variable ~ design.2$Location, se)
centroids.sd <- cast(L.centroids, variable ~ design.2$Location, sd)

cent.dataframe <- t(data.frame(rbind(centroids[1,-1], centroids[2,-1],
                             centroids.sd[1,-1],centroids.sd[2,-1])))
colnames(cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
cent.treats <- rownames(cent.dataframe)
                                  
df <- as.data.frame(cent.dataframe)
p <- ggplot(df, aes(x=V1, y=V2, colour=cent.treats)) + theme_bw() 
p + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
          axis.line = element_line(colour = "black")) + 
theme(panel.background = element_blank()) + 
  geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.01), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.01), colour="black") + 
  geom_point(size=5) +
  xlab(paste("PCoA Axis 1 (", explainvar1c, "%)", sep = "")) + 
  ylab(paste("PCoA Axis 2 (", explainvar2c, "%)", sep = "")) +  
  scale_color_manual(name="Location", values=c("#333333", "#3399FF","#66CC00" ), 
                     labels = c("Epibionts", "NO_Sargassum","Sargassum"))
```