---
title: "Gulf Stream Microbes"
author: "Ariane L. Peralta, Mario E. Muscarella, ..."
date: "`r format(Sys.time(), '%d %B, %Y')`"
header-includes:
   - \usepackage{array}
   - \usepackage{graphics}
output: 
  pdf_document:
    fig_caption: true
---

# Project Description:

# Initial Setup
```{r results='hide', message=FALSE}
rm(list=ls()) 
setwd("~/GitHub/GulfStreamMicrobiomes/analyses/")

# Simple Functions
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
sem <- function(x){sd(na.omit(x))/sqrt(length(na.omit(x)))}
CV <- function(x, ...){(sd(x, na.rm = TRUE)/mean(x, na.rm = TRUE))*100}

# Source Code
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")
# Save Default Plot Settings
opar <- par(no.readonly = TRUE)  # Saves plot defaults

# Required Packages
require("vegan"); require("stats");require("RColorBrewer")
require("png"); require("grid");require("reshape"); require("car")
```

# Import Data File
```{r}
design <- read.csv("../data/GS_design.csv")

otu.in <- read.otu("../data/phylotype/GS.trim.contigs.trim.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.tx.shared", cutoff = "1")

tax.in <- read.tax("../data/phylotype/GS.trim.contigs.trim.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.tx.1.cons.taxonomy", format = "rdp", tax.levels = 6)


tax_raw <- read.delim("../data/phylotype/GS.trim.contigs.trim.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.tx.1.cons.taxonomy", header = F, skip = 1) 

```

# OTU Data Wrangling
```{r}
# Remove OTUs with less than two occurences across all sites
otus <- otu.in[, which(colSums(otu.in) >= 2)]

# Make Presence Absence Matrix
dataPA <- (otus > 0) * 1

# Make Relative Abundence Matrices
dataREL <- otus
for(i in 1:dim(otus)[1]){
  dataREL[i,] <- otus[i,]/sum(otus[i,])
}

# Log Transform Relative Abundances
dataREL.log <- decostand(dataREL, method="log")
```

# Alpha Diversity
```{r}


```

# Beta Diversity 
```{r}

```